x/attribute/keeper/keeper.go

```diff
@@ -383,38 +383,42 @@ func (k Keeper) DeleteAttribute(ctx sdk.Context, addr string, name string, value
 
 	store := ctx.KVStore(k.storeKey)
 	it := sdk.KVStorePrefixIterator(store, types.AddrStrAttributesNameKeyPrefix(addr, name))
-	defer it.Close()
-	var count int
+	attrToDelete := []types.Attribute{} // do delete logic outside of iterator
 	for ; it.Valid(); it.Next() {
 		attr := types.Attribute{}
 		if err := k.cdc.Unmarshal(it.Value(), &attr); err != nil {
 			return err
 		}
 
 		if attr.Name == name && (!deleteDistinct || bytes.Equal(*value, attr.Value)) {
-			count++
-			store.Delete(it.Key())
-			k.DecAttrNameAddressLookup(ctx, attr.Name, attr.GetAddressBytes())
-			k.deleteAttributeExpireLookup(store, attr)
+			attrToDelete = append(attrToDelete, attr)
+		}
+	}
+	it.Close()
```

Comment:
Because there is a return on err statement above this close might not get called.  This should be a defer.

Reply:
Good catch. Fixed.